### PR TITLE
Move player skull check to lua

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Move player skull check to lua
- Move prevent player to generate loot to lua
- Improve code from player death.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by omarcopires** [comment]
That is great!!!
Perhaps you could include #3385 along with these changes.

File: src/luascript.cpp [hunk 1]
```diff
@@ -2656,6 +2654,8 @@ void LuaScriptInterface::registerFunctions()
 	registerMethod("Player", "getGuildNick", LuaScriptInterface::luaPlayerGetGuildNick);
 	registerMethod("Player", "setGuildNick", LuaScriptInterface::luaPlayerSetGuildNick);
 
+	registerMethod("Player", "isInWar", LuaScriptInterface::luaPlayerIsInWar);
```
Comment (ranisalt) [inline src/luascript.cpp:2657]: ARE AT WAR

please fix this longstanding mistake

Comment (ramon-bernardo) [reply src/luascript.cpp]: Fixed

**Comment by ramon-bernardo** [comment]
@omarcopires I added both pvp systems. with "new" and "old" mode.
image

File: data/creaturescripts/scripts/player_death.lua [hunk 2]
```diff
@@ -59,28 +59,54 @@ function onDeath(player, corpse, killer, mostDamageKiller, lastHitUnjustified, m
...
+	end
+end
+
+function onDeath(player, corpse, killer, mostDamageKiller, lastHitUnjustified, mostDamageUnjustified)
+	player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You are dead.")
+
+	local killedByPlayer, killerName = getKiller(killer)
+	local killedByPlayerMostDamage, killerNameMostDamage = getKiller(mostDamageKiller)
+
+	takeScreenShot(killedByPlayer, mostDamageKiller)
```
Comment (ranisalt) [inline data/creaturescripts/scripts/player_death.lua]: `takeScreenshot`

"screen shot" is putting a bullet through your screen

File: data/scripts/creaturescripts/player/skull_check.lua [hunk 3]
```diff
@@ -97,7 +97,7 @@ function event.onKill(player, target, lastHit)
 		return true
 	end
 
-	if not player:isInWar(targetPlayer) then
+	if not player:IsAtWarAgainst(targetPlayer) then
```
Comment (ranisalt) [inline data/scripts/creaturescripts/player/skull_check.lua:188]: Start with lowercase

Comment (ramon-bernardo) [reply data/scripts/creaturescripts/player/skull_check.lua]: Thanks

File: data/creaturescripts/scripts/player_death.lua (hunk 2 above)
Comment (ramon-bernardo) [reply data/creaturescripts/scripts/player_death.lua]: ops

**Review by Codinablack — COMMENTED** [review]
Overall looks good, but I haven't tested..

File: data/creaturescripts/scripts/player_death.lua [hunk 4]
```diff
@@ -18,33 +19,32 @@ local function getKiller(killer)
...
 	if not deathListEnabled then
 		return
 	end
 
 	local playerGuid = player:getGuid()
-	db.query("INSERT INTO `player_deaths` (`player_id`, `time`, `level`, `killed_by`, `is_player`, `mostdamage_by`, `mostdamage_is_player`, `unjustified`, `mostdamage_unjustified`) VALUES (" .. playerGuid .. ", " .. os.time() .. ", " .. player:getLevel() .. ", " .. db.escapeString(killerName) .. ", " .. (byPlayer and 1 or 0) .. ", " .. db.escapeString(killerNameMostDamage) .. ", " .. (byPlayerMostDamage and 1 or 0) .. ", " .. (lastHitUnjustified and 1 or 0) .. ", " .. (mostDamageUnjustified and 1 or 0) .. ")")
+	db.query("INSERT INTO `player_deaths` (`player_id`, `time`, `level`, `killed_by`, `is_player`, `mostdamage_by`, `mostdamage_is_player`, `unjustified`, `mostdamage_unjustified`) VALUES (" .. playerGuid .. ", " .. os.time() .. ", " .. player:getLevel() .. ", " .. db.escapeString(killerName) .. ", " .. (killedByPlayer and 1 or 0) .. ", " .. db.escapeString(killerNameMostDamage) .. ", " .. (killedByPlayerMostDamage and 1 or 0) .. ", " .. (lastHitUnjustified and 1 or 0) .. ", " .. (mostDamageUnjustified and 1 or 0) .. ")")
```
[the remark this answers is not in the document]
Comment (ramon-bernardo) [reply data/creaturescripts/scripts/player_death.lua:47]: this is a local variable, renamed from byPlayer to killedByPlayer. SQL is the same.

**Pull request closed without merging**